### PR TITLE
[conflict] Implicitly assign class type to the first parameter of its methods (#1514)

### PR DESCRIPTION
**Pull request by thiago-silva**
Follow up on PR #1586: assign the class type to first parameter of it's dynamic method if --check-untyped-defs is enabled and if the parameter variable is named "self".

### CODE REVIEW

**Comment by elazarg** [comment]
I'm afraid this PR might be in some conflict with #2193.

**Comment by gvanrossum** [comment]
Can you rebase this?

File: mypy/nodes.py [hunk 1]
```diff
@@ -2244,8 +2246,16 @@ def function_type(func: FuncBase, fallback: 'mypy.types.Instance') -> 'mypy.type
         if name:
             name = '"{}"'.format(name)
 
+        if options.check_untyped_defs and len(fdef.arguments) > 0 and\
```
Comment (gvanrossum) [inline mypy/nodes.py]: Please use () to wrap long lines, not \.

**Comment by gvanrossum** [comment]
I wonder iif there is a more economic way to get the option to the place that needs it?

**Comment by gvanrossum** [comment]
Hey @thiago-silva, this PR has been without activity, waiting for a conflict resolution, for over 6 weeks. I am going to close it now; if you still want to work on it, just open a new PR and we'll review it!

Note: it's also possible that this idea is simply superseded by "self types" (#2193), but I'm not sure.

**Pull request closed without merging**